catroid/src/org/catrobat/catroid/formulaeditor/SensorHandler.java

```diff
@@ -117,50 +155,92 @@ public static Double getSensorValue(Sensors sensor) {
 		switch (sensor) {
 
 			case X_ACCELERATION:
-				return (double) instance.linearAcceleartionX;
+				return (double) instance.linearAccelerationX;
 
 			case Y_ACCELERATION:
-				return (double) instance.linearAcceleartionY;
+				return (double) instance.linearAccelerationY;
 
 			case Z_ACCELERATION:
-				return (double) instance.linearAcceleartionZ;
+				return (double) instance.linearAccelerationZ;
 
 			case COMPASS_DIRECTION:
 				float[] orientations = new float[3];
-				android.hardware.SensorManager.getRotationMatrixFromVector(instance.rotationMatrix,
-						instance.rotationVector);
+				if (!instance.useRotationVectorFallback) {
```

Comment:
what happens here if useRotationVectorFallback is true? correct me if I'm wrong, but if there is no getRotationMatrixFromVector then the rotationMatrix is not initialised and hence you won't get any reasonable results anyway.